Add Versioning

Cahllagerfeld opened the pull request:
closes #2247

https://deploy-preview-2725--gitpod-kumquat.netlify.app/

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2725"><img src="https://gitpod-staging.com/button/open-in-gitpod.svg"/></a>



<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2725"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

Siddhant-K-code commented:
This PR is so huge. I'm just worried about the Review & potential breaking changes.
Also, there will be a lot of conflict due to #2587

When are you planning to Ship this?

Cahllagerfeld commented:
Planned early next week

Cahllagerfeld commented:
Hey @lucasvaltl, 
thats the draft PR for versioning the entire docs. I added a dummy version for testing purposes.
Things I did:
- Make links relative
- removed `latest` tag in self-hosted docs
- make left-side-menu dynamic

Note: The build time increased from about 4 min to 6 min.

‼️ Before merging we need to revert the last commit.

Cahllagerfeld commented on gitpod/docs/configure/self-hosted/latest/reference-architecture/_chunks/cluster.md:
```diff
@@ -6,21 +6,17 @@ layout: false
   import CloudPlatformToggle from "$lib/components/docs/cloud-platform-toggle.svelte";
```
Double check these files

Cahllagerfeld commented:
closing this issue for now, because it's on hold for a long time now
